src/Microsoft.ML.Core/Data/SchemaBuilder.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// A builder for <see cref="Schema"/>.
+    /// </summary>
+    public sealed class SchemaBuilder
+    {
+        private readonly List<(string Name, ColumnType Type, Schema.Metadata Metadata)> _items;
+
+        /// <summary>
+        /// Create a new instance of <see cref="SchemaBuilder"/>.
+        /// </summary>
+        public SchemaBuilder()
```

Comment:
>SchemaBuilder [](start = 15, length = 13)

I've look into your code, and everywhere pattern for this thing is following:
AddColumns to builder.
Call GetSchema.
We done.

Why we even need this object? 
I would understand it necessity if you had cases of:
AddColumns
GetSchema
AddMoreColumns
GetAnotherSchema,
but I don't see it.

Can we kill it and let Schema to be have constructor which accepts IEnumerable/IList of columns?

Reply:
Let's keep it as is for now. 1) it's similar to `MetadataBuilder`, and it's better to have one pattern than two. 2) It's common to build the schema somewhat incrementally, and builder is a more convenient pattern than maintaining a collection. 2) We can, in the future, move builders to a separate namespaces, since they are not intended for end users.

---
In reply to: [235181853](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235181853) [](ancestors = 235181853)

Reply:
Voice of my inner monk with a blade are in disagree, but I wouldn't advice to listen to voices in my head, especially if they are armed.

---
In reply to: [235477101](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235477101) [](ancestors = 235477101,235181853)